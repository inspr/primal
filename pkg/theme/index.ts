import { Unpack } from "@primal/state"
import Theme from "./src/theme"

// Apply the theme definitions to the global scope
// We need this definition to primal recognize the custom props
declare module "@primal/primitives" {
    /**
     * @type Inspr
     * @extends Spec
     * `@primal/theme` is currently applied as theme the app.
     *
     * The theme is designed and managed by inspr team and it's based on inspr design language
     * @link https://inspr.design/theme
     */

    type Theme = Unpack<typeof Theme>["__theme"]
    interface PrimitivesTheme extends Theme {}
}

export default Theme
