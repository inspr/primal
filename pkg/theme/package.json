{
    "name": "@primal/theme",
    "version": "0.0.0",
    "description": "",
    "main": "./index.ts",
    "types": "./index.ts",
    "dependencies": {
        "@types/next": "^9.0.0",
        "@types/react": "^17.0.39"
    }
}
