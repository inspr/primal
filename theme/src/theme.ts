import { createTheme } from "@primal/primitives"

const theme = createTheme({
    token: {
        color: {
            swatch: {
                brown: {
                    100: "#f8f3f1",
                    200: "#e4d0c9",
                    300: "#cfada0",
                    400: "#bb8977",
                    500: "#a36852",
                    600: "#7a4e3e",
                    700: "#513429",
                    800: "#36231b",
                    900: "#1b120e",
                },
                yellow: {
                    100: "#fff8eb",
                    200: "#ffe4ad",
                    300: "#ffcf70",
                    400: "#ffba33",
                    500: "#f5a300",
                    600: "#b87a00",
                    700: "#7a5200",
                    800: "#523600",
                    900: "#241800",
                },
                orange: {
                    100: "#fff1eb",
                    200: "#ffc9ad",
                    300: "#ffa070",
                    400: "#ff7733",
                    500: "#f55200",
                    600: "#b83d00",
                    700: "#7a2900",
                    800: "#521b00",
                    900: "#290d00",
                },
                purple: {
                    100: "#f5ebff",
                    200: "#d6adff",
                    300: "#b870ff",
                    400: "#9933ff",
                    500: "#7a00f5",
                    600: "#5c00b8",
                    700: "#3d007a",
                    800: "#290052",
                    900: "#140029",
                },
                red: {
                    100: "#ffebee",
                    200: "#ffadbb",
                    300: "#ff7088",
                    400: "#ff3355",
                    500: "#e50026",
                    600: "#b8001f",
                    700: "#7a0014",
                    800: "#52000e",
                    900: "#290007",
                },
                pink: {
                    100: "#feecf8",
                    200: "#fab2e2",
                    300: "#f679cd",
                    400: "#f231b2",
                    500: "#e60f9e",
                    600: "#ad0b77",
                    700: "#73074f",
                    800: "#4d0535",
                    900: "#26021a",
                },
                blue: {
                    100: "#ecf2fe",
                    200: "#b2cafb",
                    300: "#78a2f7",
                    400: "#3e7bf4",
                    500: "#0d56e7",
                    600: "#0a41ae",
                    700: "#072b74",
                    800: "#041d4d",
                    900: "#020f27",
                },
                green: {
                    100: "#effaf1",
                    200: "#c1ecc6",
                    300: "#92dd9b",
                    400: "#63cf70",
                    500: "#3abb49",
                    600: "#2b8c36",
                    700: "#1d5e24",
                    800: "#133e18",
                    900: "#0a1f0c",
                },
                gray: {
                    50: "#fefefe",
                    100: "#f5f5f5",
                    200: "#d6d6d6",
                    300: "#b8b8b8",
                    400: "#999999",
                    500: "#7a7a7a",
                    600: "#5c5c5c",
                    700: "#3d3d3d",
                    800: "#292929",
                    900: "#111111",
                },
            },
            kind: {
                lumen: "gray.50",
                umbra: "gray.900",
                creativity: "yellow.400",
                danger: "red.500",
                privacy: "purple.400",
                trust: "blue.500",
            },
            gradient: {
                cta: "linear-gradient(90deg, #fc5d32 0%, #ee4c54 25%, #d45e95 50%, #9c6ca6 75%, #6583c1 100%)",
            },
        },
        font: {
            family: {
                inter: '"Inter", sans-serif',
                opensauce: '"Open Sauce One", sans-serif',
            },
            weight: {
                regular: "400",
                medium: "500",
                semibold: "600",
                bold: "700",
            },
            letterSpace: {
                sm: "-0.02rem",
                md: "0.01rem",
                lg: "0.04rem",
            },
            lineHeight: {
                xxxs: "12px",
                xxs: "16px",
                xs: "20px",
                sm: "24px",
                md: "28px",
                lg: "36px",
                xl: "44px",
                xxl: "64px",
                xxxl: "80px",
            },
            size: {
                xxxs: "10px",
                xxs: "12px",
                xs: "14px",
                sm: "16px",
                md: "18px",
                lg: "28px",
                xl: "36px",
                xxl: "48px",
                xxxl: "64px",
            },
        },
        shadow: {
            xs: "0px 2px 4px rgba(17, 17, 17, 0.04), 0px 1px 2px rgba(17, 17, 17, 0.04)",
            sm: "0px 2px 4px rgba(17, 17, 17, 0.05), 0px 2px 4px rgba(17, 17, 17, 0.05)",
            md: "0px 4px 6px rgba(17, 17, 17, 0.04), 0px 6px 12px 2px rgba(17, 17, 17, 0.05)",
            lg: "0px 16px 32px 2px rgba(17, 17, 17, 0.08), 0px 32px 48px rgba(17, 17, 17, 0.08)",
            xl: "0px 48px 64px 2px rgba(17, 17, 17, 0.1), 0px 64px 80px rgba(17, 17, 17, 0.1)",
        },
        size: {
            xxxs: "2px",
            xxs: "4px",
            xs: "8px",
            sm: "16px",
            md: "24px",
            lg: "32px",
            xl: "48px",
            xxl: "64px",
            xxxl: "96px",
        },
        breakpoint: {
            xs: { min: "0px", max: "600px" },
            sm: { min: "600px", max: "900px" },
            md: { min: "900px", max: "1280px" },
            lg: { min: "1280px", max: "1920px" },
            xl: { min: "1920px", max: "10000px" },
        },
    },
    variants: {
        text: {
            default: {
                fontStyle: "normal",
                fontVariant: "normal",
                fontWeight: "400",
                fontSize: "16px",
                lineHeight: "24px",
                fontFamily: '"Inter", sans-serif',
            },
            caption: {
                fontStyle: "normal",
                fontVariant: "normal",
                fontWeight: "400",
                fontSize: "12px",
                lineHeight: "16px",
                fontFamily: '"Inter", sans-serif',
            },
            display: {
                fontStyle: "normal",
                fontVariant: "normal",
                fontWeight: "700",
                fontSize: "48px",
                lineHeight: "44px",
                fontFamily: '"Open Sauce One", sans-serif',
            },
            overline: {
                fontStyle: "normal",
                fontVariant: "normal",
                fontWeight: "600",
                fontSize: "12px",
                lineHeight: "16px",
                fontFamily: '"Inter", sans-serif',
            },
            subtitle: {
                fontStyle: "normal",
                fontVariant: "normal",
                fontWeight: "500",
                fontSize: "16px",
                lineHeight: "16px",
                fontFamily: '"Inter", sans-serif',
            },
            title: {
                fontStyle: "normal",
                fontVariant: "normal",
                fontWeight: "700",
                fontSize: "28px",
                lineHeight: "36px",
                fontFamily: '"Open Sauce One", sans-serif',
            },
        },
        button: {
            primary: {
                textColor: "gray.50",
                backgroundColor: "gray.900",
                borderColor: "gray.900",
                iconColor: "invert(1)",
                hover: {
                    textColor: "gray.900",
                    backgroundColor: "gray.50",
                    iconColor: "invert(0)",
                },
            },
            secondary: {
                textColor: "gray.900",
                backgroundColor: "gray.50",
                borderColor: "gray.900",
                hover: {
                    textColor: "gray.50",
                    backgroundColor: "gray.900",
                    iconColor: "invert(1)",
                },
            },
            success: {
                textColor: "gray.50",
                backgroundColor: "blue.500",
                borderColor: "blue.500",
                iconColor: "invert(1)",
                hover: {
                    textColor: "blue.500",
                    backgroundColor: "gray.50",
                    iconColor:
                        "invert(17%) sepia(100%) saturate(4267%) hue-rotate(221deg) brightness(97%) contrast(90%)",
                },
            },
            error: {
                textColor: "gray.50",
                backgroundColor: "red.500",
                borderColor: "red.500",
                iconColor: "invert(1)",
                hover: {
                    textColor: "red.500",
                    backgroundColor: "gray.50",
                    iconColor:
                        "invert(24%) sepia(50%) saturate(6429%) hue-rotate(338deg) brightness(81%) contrast(128%)",
                },
            },
            disabled: {
                textColor: "gray.600",
                backgroundColor: "gray.400",
                borderColor: "gray.400",
                iconColor:
                    "invert(38%) sepia(1%) saturate(0%) hue-rotate(42deg) brightness(90%) contrast(88%)",
            },
            cta: {
                textColor: "lumen",
                backgroundColor: "cta",
                borderColor: "transparent",
                shadow: "md",
                iconColor: "invert(1)",
                hover: {
                    shadow: "lg",
                    opacity: {
                        value: 0.35,
                        color: "umbra",
                    },
                },
            },
        },
        switch: {
            primary: {
                onColor: "umbra",
                offColor: "gray.200",
                ballColor: "gray.50",
            },
            secondary: {
                onColor: "gray.500",
                offColor: "gray.200",
                ballColor: "gray.50",
            },
            success: {
                onColor: "blue.500",
                offColor: "gray.200",
                ballColor: "gray.50",
            },
            error: {
                onColor: "danger",
                offColor: "gray.200",
                ballColor: "gray.50",
            },
        },
        checkbox: {
            primary: {
                backgroundColor: "gray.900",
                borderColor: "gray.900",
                checked: {
                    backgroundColor: "lumen",
                    checkColor: "umbra",
                },
            },
            secondary: {
                backgroundColor: "lumen",
                borderColor: "umbra",
                checked: {
                    backgroundColor: "umbra",
                    checkColor: "lumen",
                },
            },
            success: {
                backgroundColor: "blue.500",
                borderColor: "blue.500",
                checked: {
                    backgroundColor: "lumen",
                    checkColor: "blue.500",
                },
            },
            error: {
                backgroundColor: "red.500",
                borderColor: "red.500",
                checked: {
                    backgroundColor: "lumen",
                    checkColor: "red.500",
                },
            },
        },
        radio: {
            primary: {
                textColor: "umbra",
                backgroundColor: "gray.900",
                borderColor: "gray.900",
            },
            success: {
                textColor: "umbra",
                backgroundColor: "blue.500",
                borderColor: "blue.500",
            },
            error: {
                textColor: "umbra",
                backgroundColor: "red.500",
                borderColor: "red.500",
            },
            disabled: {
                textColor: "gray.600",
                backgroundColor: "gray.600",
                borderColor: "gray.600",
            },
        },
    },
})

export default theme
