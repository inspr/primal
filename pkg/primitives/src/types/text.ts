import { Token } from "../theme"
import { ColorOptions } from "./"
import { TextTypes } from "./variants" 

type FontFamilyValues = keyof Token["font"]["family"]
type LineHeightValues = keyof Token["font"]["lineHeight"]
type FontSizeValues = keyof Token["font"]["size"]
type LetterSpaceValues = keyof Token["font"]["letterSpace"]
type FontWeightValues = keyof Token["font"]["weight"]
type FontStyle = "serif" | "sans-serif"
type WhiteSpaceOptions = "normal" | "nowrap" | "pre-line"
type TextAlignOptions = "left" | "right" | "center" | "justify" | "start"
type TextDecorationLine = "none" | "underline"

export type FontFamily = `"${string}", ${FontStyle}`
export type FontStyleOptions =
    | "normal"
    | "italic"
    | "oblique"
    | "initial"
    | "inherit"

export type FontVariantOptions = "normal" | "small-caps" | "initial" | "inherit"
export type InternalHRef = `/${string}`
export type HRef = `https://${string}` | InternalHRef

interface TextHoverEffect {
    textDecoration?: TextDecorationLine
}

export interface TextProps {
    color?: ColorOptions
    textAlign?: TextAlignOptions
    whiteSpace?: WhiteSpaceOptions
    textVariant?: TextTypes
    family?: FontFamilyValues
    weight?: FontWeightValues
    letterSpace?: LetterSpaceValues
    lineHeight?: LineHeightValues
    size?: FontSizeValues
    uppercase?: boolean
    italic?: boolean
    href?: HRef
    textDecoration?: TextDecorationLine
    fitContent?: boolean
    hover?: TextHoverEffect
    __unsafeClassName?: string
}
