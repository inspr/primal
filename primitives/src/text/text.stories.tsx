import { ComponentMeta, ComponentStory } from "@storybook/react"
import React from "react"
import { Text } from "./text"

export default {
    title: "Typography/Text",
    component: Text,
} as ComponentMeta<typeof Text>

const Template: ComponentStory<typeof Text> = (args) => (
    <Text {...args}>This is an example</Text>
)

export const Display = Template.bind({})
Display.args = {
    textVariant: "display",
}

export const Title = Template.bind({})
Title.args = {
    textVariant: "title",
}

export const Subtitle = Template.bind({})
Subtitle.args = {
    textVariant: "subtitle",
}

export const Default = Template.bind({})
Default.args = {
    textVariant: "default",
}

export const Caption = Template.bind({})
Caption.args = {
    textVariant: "caption",
}

export const Overline = Template.bind({})
Overline.args = {
    textVariant: "overline",
    uppercase: true,
}
