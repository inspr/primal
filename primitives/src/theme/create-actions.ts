import { AnySize, Swatch, VariantOptions, Weight } from "../types"
import { Spec, Token } from "./index"

//TODO check types that are "string | number" e.g "presetValueAux"

export const createActions = (theme: Spec) => ({
    color(colorValue: string | `${string}.${Weight}`) {
        if (!colorValue || colorValue === "transparent") return colorValue

        const { swatch, kind, gradient } = theme.token.color

        if (Object.keys(gradient).includes(colorValue)) {
            return gradient[colorValue]
        }

        let colorValueAux = colorValue

        if (Object.keys(kind).includes(colorValue)) {
            colorValueAux = kind[colorValue]
        }
        const [color, weight] = colorValueAux.split(".") as [
            keyof typeof swatch,
            Weight
        ]
        const selectedColor: { [weight in Weight]?: Swatch } = swatch[color]
        // TODO: fix typing so this validation isn't necessary (ex: brown.50 seems to be valid)
        const validColor = selectedColor && selectedColor[weight]
        return validColor ? selectedColor[weight]! : "transparent"
    },
    fontWeight(weightValue: string) {
        const { weight } = theme.token.font
        let weightOptions = weightValue as keyof typeof weight
        return weight[weightOptions]
    },
    fontFamily(familyValue: string) {
        const { family } = theme.token.font
        let familyOptions = familyValue as keyof typeof family
        return family[familyOptions]
    },
    fontSize(sizeValue: AnySize) {
        const { size } = theme.token.font
        let sizeOptions = sizeValue as keyof typeof size
        return size[sizeOptions]
    },
    letterSpace(spaceValue: AnySize) {
        const { letterSpace } = theme.token.font
        // const letterSpace = theme.token.font["letterSpace"]
        let spaceOptions = spaceValue as keyof typeof letterSpace
        return letterSpace[spaceOptions]
    },
    lineHeight(heightValue: AnySize) {
        const { lineHeight } = theme.token.font
        let heightOptions = heightValue as keyof typeof lineHeight
        return lineHeight[heightOptions]
    },
    size(sizeValue: AnySize) {
        const { size } = theme.token
        const sizeOptions = sizeValue as keyof typeof size
        return size[sizeOptions]
    },
    shadow(shadowValue: AnySize) {
        const { shadow } = theme.token
        const shadowOptions = shadowValue as keyof typeof shadow
        return shadow[shadowOptions]
    },
    breakpoints() {
        return theme.token.breakpoint as Token["breakpoint"]
    },
    variants(component: keyof VariantOptions, variant: string) {
        const { variants } = theme
        const selected = variants[component]
        const type = variant as keyof typeof selected

        return variants[component][type]
    },
})

export type ActionType = ReturnType<typeof createActions>
