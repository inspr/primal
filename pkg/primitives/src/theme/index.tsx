import type { PrimitivesTheme } from "@primal/primitives"
import { AtomicState, createState, Unpack, useInspr } from "@primal/state"
import {
    AnyShadow,
    AnyTypedSize,
    Color,
    FontFamily,
    PX,
    Size,
    Unit,
    Weight,
} from "../types"
import { VariantOptions } from "../types/variants"
import { createActions } from "./create-actions"

interface DefaultTheme {
    token: {
        shadow: { [size in Size]: string }
        color: {
            swatch: {}
            kind: {}
            gradient: {}
        }
        font: {
            family: {}
            lineHeight: {}
            weight: {}
            letterSpace: {}
            size: {}
        }
        size: {}
        breakpoint: {}
    }
    variants: {}
}

type Theme = DefaultTheme & PrimitivesTheme
export type Token = Theme["token"]
export type Variants = Theme["variants"]

export interface Spec {
    token: {
        color: Color
        font: {
            family: {
                [key: string]: FontFamily
            }
            weight: {
                [key: string]: Weight
            }
            letterSpace: {
                [size in Size<0>]: Unit
            }
            lineHeight: AnyTypedSize<Unit>
            size: AnyTypedSize<PX>
        }
        shadow: AnyTypedSize<AnyShadow>
        size: AnyTypedSize<Unit>
        breakpoint: AnyTypedSize<{ min: PX; max: PX }>
    }
    variants: VariantOptions
}

let primitivesThemeContext: AtomicState<any>

export const createTheme = <T extends Spec>(spec: T) => {
    const theme = createState({
        __theme: spec,
        actions: createActions(spec),
    })

    primitivesThemeContext = theme
    return theme
}

export const useTheme = (): Unpack<ReturnType<typeof createTheme>> => {
    if (!primitivesThemeContext) {
        if (process.env.NODE_ENV === "development") {
            throw new Error("A theme must be registered before use")
        }
    }
    return useInspr(primitivesThemeContext)
}
