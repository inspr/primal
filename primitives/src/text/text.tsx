import { css, SerializedStyles } from "@emotion/react"
import styled from "@emotion/styled"
import React, { forwardRef } from "react"
import { useTheme } from "../theme"
import { ActionType } from "../theme/create-actions"
import { TextProps, TextTypes, TextVariant } from "../types"
import { BaseProps } from "../utils"


const generateTextVariantStyles = (
    actions: ActionType,
    variant: TextTypes,
) => {
    const text = actions.variants(
        "text",
        variant
    ) as TextVariant[typeof variant]

    return css`
        ${text.fontStyle ? `font-style: ${text.fontStyle}` : ""};
        ${text.fontVariant ? `font-variant: ${text.fontVariant}` : ""};
        ${text.fontWeight ? `font-weight: ${text.fontWeight}` : ""};
        ${text.fontSize ? `font-size: ${text.fontSize}` : ""};
        ${text.lineHeight ? `line-height: ${text.lineHeight}` : ""};
        ${text.fontFamily ? `font-family: ${text.fontFamily}` : ""};
    `
}

const TextStyle = styled.a<
    {
        actions: ActionType
        textVariant?: TextTypes
        injectedCSS?: SerializedStyles
    } & TextProps
>`
    ${(props) => 
        props.textVariant
            ? generateTextVariantStyles(props.actions, props.textVariant) 
            : ""}
    color: ${(props) => props.actions.color(props.color!)};
    ${(props) => (props.textAlign ? `text-align: ${props.textAlign};` : "")}
    ${(props) => (props.whiteSpace ? `white-space: ${props.whiteSpace};` : "")}
    ${(props) =>
        props.weight
            ? `font-weight: ${props.actions.fontWeight(props.weight)};`
            : ""}
    ${(props) =>
        props.family
            ? `font-family: ${props.actions.fontFamily(props.family)};`
            : ""}
    ${(props) =>
        props.size ? `font-size: ${props.actions.fontSize(props.size)};` : ""}
    ${(props) =>
        props.letterSpace
            ? `letter-spacing: ${props.actions.letterSpace(props.letterSpace)};`
            : ""}
    ${(props) =>
        props.lineHeight
            ? `line-height: ${props.actions.lineHeight(props.lineHeight)};`
            : ""}
    text-transform: ${(props) => (props.uppercase ? "uppercase" : "none")};
    font-style: ${(props) => (props.italic ? "italic" : "normal")};
    text-decoration: ${(props) => props.textDecoration};
    ${(props) => (props.fitContent ? "width: fit-content;" : "")}

    &:hover {
        ${(props) =>
            props.hover?.textDecoration
                ? `text-decoration: ${props.hover.textDecoration};`
                : ""}
    }
    ${(props) => (props.injectedCSS ? props.injectedCSS : "")}
`

export type TextKinds =
    | "h1"
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6"
    | "p"
    | "i"
    | "u"
    | "span"
    | "a"
    | "li"

const Text = forwardRef<any, BaseProps<TextKinds> & TextProps>(
    (
        {
            is = "span",
            color = "gray.900",
            textVariant = "default",
            textAlign,
            whiteSpace,
            weight,
            size,
            letterSpace,
            family,
            lineHeight,
            uppercase,
            italic,
            textDecoration = "none",
            fitContent = false,
            hover,
            children,
            href,
            injectedCSS,
            __unsafeClassName,
        },
        ref
    ) => {
        const C: React.ElementType = is

        const { actions } = useTheme()

        return (
            <TextStyle
                as={is}
                ref={ref}
                href={href}
                className={__unsafeClassName}
                color={color}
                actions={actions}
                injectedCSS={injectedCSS}
                textVariant={textVariant}
                textAlign={textAlign}
                whiteSpace={whiteSpace}
                weight={weight}
                size={size}
                letterSpace={letterSpace}
                family={family}
                lineHeight={lineHeight}
                uppercase={uppercase}
                italic={italic}
                textDecoration={textDecoration}
                fitContent={fitContent}
                hover={hover}>
                {children}
            </TextStyle>
        )
    }
)

Text.displayName = "Text"

export { Text }
