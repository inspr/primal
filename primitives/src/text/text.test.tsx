import "@primal/theme"
import { create } from "react-test-renderer"
import { Text } from "./text"

describe("Text", () => {
    it("renders a custom Text", () => {
        const view = create(
            <Text
                is="p"
                textVariant="caption"
                textAlign="center"
                whiteSpace="pre-line"
                weight="bold"
                family="opensauce"
                size="xl"
                letterSpace="sm"
                lineHeight="xxl"
                uppercase
                italic
                color="blue.400">
                Some Text
            </Text>
        )
        const props = view.root.findByProps({ is: "p" }).props
        expect(props).toHaveProperty("children", "Some Text")
        expect(props).toHaveProperty("textVariant", "caption")
        expect(props).toHaveProperty("textAlign", "center")
        expect(props).toHaveProperty("whiteSpace", "pre-line")
        expect(props).toHaveProperty("weight", "bold")
        expect(props).toHaveProperty("family", "opensauce")
        expect(props).toHaveProperty("size", "xl")
        expect(props).toHaveProperty("letterSpace", "sm")
        expect(props).toHaveProperty("lineHeight", "xxl")
        expect(props).toHaveProperty("uppercase", true)
        expect(props).toHaveProperty("italic", true)
        expect(props).toHaveProperty("color", "blue.400")
    })

    it("should match snapshot", () => {
        const text = create(<Text>Some Text</Text>)
        expect(text.toJSON()).toMatchSnapshot()
    })
})
