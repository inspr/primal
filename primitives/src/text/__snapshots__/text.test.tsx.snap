// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match snapshot 1`] = `
<span
  className="css-1h6u694"
  color="gray.900"
  textDecoration="none"
>
  Some Text
</span>
`;
